apps/communications/dialer/test/unit/keypad_test.js

```diff
@@ -166,6 +167,31 @@ suite('dialer/keypad', function() {
         };
       });
 
+      test('Properly highlight the last key in an abbreviated dialing code',
+      function() {
+        for (var i = 0, end = mmi.length; i < end; i++) {
+          fakeEventStart.target.dataset.value = mmi.charAt(i);
+          subject.keyHandler(fakeEventStart);
+          assert.isTrue(node.classList.contains('active'));
+          fakeEventEnd.target.dataset.value = mmi.charAt(i);
+          subject.keyHandler(fakeEventEnd);
+          assert.isFalse(node.classList.contains('active'));
+        }
+      });
+
+      test('Start an abbreviated dialing code operation only upon pressing #',
+      function() {
+        this.sinon.spy(KeypadManager, '_getSpeedDialNumber');
+        subject._phoneNumber = '1#1';
+
```

Comment:
Yes, what this does is have '1#1' appear to have been already typed and delete the final '1' so we end up with '1#'. Without my changes this test triggers the speed dial action and fails.